### ISSUE TITLE
Manage Reusable blocks without leaving Gutenberg

### ISSUE THREAD

**Issue by paaljoachim** [WordPress/gutenberg#26408]
Bringing the Manage Reusable Block into the Gutenberg screen.
Looking at the current method there is a manage reusable block link in the Inserter panel. 

Here is an example of the Reusable block inserter panel:

<img width="348" alt="Screen Shot 2020-10-22 at 23 52 13" src="https://user-images.githubusercontent.com/5323259/96933932-a9815080-14c1-11eb-9c70-8571ee8016f8.png">

Clicking **Manage all reusable blocks** 
exits the Gutenberg screen, and goes here

![Screen Shot 2020-10-22 at 22 18 38](https://user-images.githubusercontent.com/5323259/96933981-c027a780-14c1-11eb-96f4-17e678911e68.png)

It would be a lot more helpful to have editing happen inside the Gutenberg screen. Making it possible to turn on and off managing Reusable blocks. The design suggestion I made still keeps it associated with the Inserter panel. It would be very helpful to keep the association in place. Keeping managing reusable blocks close to where we already insert the block into the layout.

Clicking **Manage all reusable blocks** 
Could instead just show the Edit l Trash l Export links like so

![Screen Shot 2020-10-23 at 00 03 00](https://user-images.githubusercontent.com/5323259/96934859-42fd3200-14c3-11eb-990b-cfd2e2757ea1.png)

Opening it up directly in the same Reusable inserter panel.

Here is a prototype that I made in figma.

https://www.figma.com/proto/bV0gCaIWKHtjidtus3WwJU/Reusable-block-manage-screen?node-id=1%3A2&scaling=min-zoom

A walk through animated gif.
![Manage-Reusable-Blocks](https://user-images.githubusercontent.com/5323259/96934710-f9144c00-14c2-11eb-91d9-b3535a50f872.gif)

**Comment by paaljoachim** [comment]
Here is one way to handle trashing a reusable block. Draft would also be added here. 
Look at the flow. The placement of the All l Published l Trash options are likely not ideal. This though gives an idea of what it might look like.

Here is another animated gif.

![Trash-Restore-Reusable-Block](https://user-images.githubusercontent.com/5323259/97257855-66074900-1817-11eb-8b24-f12cf290e740.gif)

To test the prototype: https://www.figma.com/proto/bV0gCaIWKHtjidtus3WwJU/Reusable-block-manage-screen?node-id=1%3A2&scaling=min-zoom

**Comment by paaljoachim** [comment]
Next version. This time with an overlay.
An overlay can be used to show Reusable blocks, patterns, template parts, template sections etc in a bigger area.
Focus is on showing bigger versions and having an area inside Gutenberg to manage the selected.

![Manage-Reusable-Blocks-overlay-1](https://user-images.githubusercontent.com/5323259/98681656-461d6c80-2363-11eb-9358-4ad42274c782.jpg)

Clicking "Manage all reusable blocks" could just open the overlay.
The expand icon could also be seen in Patterns, as well as the other areas.

![Manage-Reusable-Blocks-overlay-2](https://user-images.githubusercontent.com/5323259/98681809-72d18400-2363-11eb-9770-013178b1b2ea.jpg)

**Comment by noisysocks** [comment]
Related to this is that one cannot remove a reusable block from the trash from within the editor.

See https://core.trac.wordpress.org/ticket/51535.

**Comment by youknowriad** [comment]
I think given how the design of patterns evolved: having a dedicated section within the site editor and being able to use them in different places (posts, pages, templates...), I think this issue is probably not relevant anymore. I'm going to close it but please let me know if you think otherwise. We might want to refresh the issue in that case.

**Comment by paaljoachim** [comment]
Yeah. This can be closed out as it is not relevant any longer.
